Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Input/DeviceManager/Controllers/OculusQuestHand.cs

```diff
@@ -0,0 +1,591 @@
+        private MixedRealityPose currentGripPose = MixedRealityPose.ZeroIdentity;
+
+        /// <summary>
+        /// Teleport pointer reference. Needs custom pointer because MRTK does not support teleporting with articulated hands.
+        /// </summary>
+        public CustomTeleportPointer TeleportPointer { get; set; }
+
+#if OCULUSINTEGRATION_PRESENT
+        private Material handMaterial = null;
+        private Renderer handRenderer = null;
+
+        private bool isIndexGrabbing = false;
+        private bool isMiddleGrabbing = false;
+        private bool isThumbGrabbing = false;
+#endif
+
+
+        private int pinchStrengthProp;
+
+        /// <summary>
+        /// Default constructor used by reflection for profiles
+        /// </summary>
+        /// <param name="trackingState"></param>
+        /// <param name="controllerHandedness"></param>
+        /// <param name="inputSource"></param>
+        /// <param name="interactions"></param>
+        public OculusQuestHand(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+            : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+            pinchStrengthProp = Shader.PropertyToID(MRTKOculusConfig.Instance.PinchStrengthMaterialProperty);
+        }
+
+        #region IMixedRealityHand Implementation
+
+        protected readonly Dictionary<TrackedHandJoint, MixedRealityPose> jointPoses = new Dictionary<TrackedHandJoint, MixedRealityPose>();
+        /// <inheritdoc/>
+        public override bool TryGetJoint(TrackedHandJoint joint, out MixedRealityPose pose)
+        {
+            return jointPoses.TryGetValue(joint, out pose);
+        }
+
+        #endregion IMixedRealityHand Implementation
+
+#if OCULUSINTEGRATION_PRESENT
+        public void InitializeHand(OVRHand ovrHand, Material handMaterial)
+        {
+            handRenderer = ovrHand.GetComponent<Renderer>();
+            UpdateHandMaterial(handMaterial);
+        }
+
+        public void UpdateHandMaterial(Material newHandMaterial)
+        {
+            if (newHandMaterial == null || !MRTKOculusConfig.Instance.UseCustomHandMaterial) return;
+
+            if (handMaterial != null)
+            {
+                Object.Destroy(handMaterial);
+            }
+            handMaterial = new Material(newHandMaterial);
+            handRenderer.sharedMaterial = handMaterial;
+        }
+
+        public void CleanupHand()
+        {
+            if (handRenderer != null)
+            {
+                handRenderer.enabled = false;
+            }
+        }
+
+        public override MixedRealityInteractionMapping[] DefaultInteractions => new[]
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer, new MixedRealityInputAction(4, "Pointer Pose", AxisType.SixDof)),
```

Comment:
Actions shouldn't be hardcoded like this. It should instead be read from the controller mapping profile

```suggestion
            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer),
```